teamcity: failed tests on master: testrace/TestQuotaPool

The following tests appear to have failed:

[#312170](https://teamcity.cockroachdb.com/viewLog.html?buildId=312170):

```
--- FAIL: testrace/TestQuotaPool (45.720s)
<autogenerated>:14: storage/client_raft_test.go:1806, condition failed to evaluate within 45s: expected 1 queued quota release, found: 6
------- Stdout: -------
I170801 23:59:42.816026 39065 storage/store.go:1271  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170801 23:59:42.820924 39065 gossip/gossip.go:295  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:49093" > attrs:<> locality:<> 
W170801 23:59:42.844652 39065 gossip/gossip.go:1204  [n2] no incoming or outgoing connections
I170801 23:59:42.857060 39065 storage/store.go:1271  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I170801 23:59:42.857423 39065 gossip/gossip.go:295  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:33153" > attrs:<> locality:<> 
I170801 23:59:42.859188 39176 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:49093
W170801 23:59:42.953331 39065 gossip/gossip.go:1204  [n3] no incoming or outgoing connections
I170801 23:59:42.960666 39317 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:49093
I170801 23:59:42.964938 39065 gossip/gossip.go:295  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:48825" > attrs:<> locality:<> 
I170801 23:59:42.974872 39441 storage/replica_raftstorage.go:509  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 52857470 at index 15
I170801 23:59:42.977307 39441 storage/store.go:3486  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 33, log entries: 5, rate-limit: 2.0 MiB/sec, 2ms
I170801 23:59:42.978617 39459 storage/replica_raftstorage.go:718  [s2,r1/?:{-}] applying preemptive snapshot at index 15 (id=52857470, encoded size=5632, 1 rocksdb batches, 5 log entries)
I170801 23:59:42.981495 39459 storage/replica_raftstorage.go:726  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I170801 23:59:42.984077 39441 storage/replica_command.go:3600  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2]
I170801 23:59:42.988922 39461 storage/replica.go:2937  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170801 23:59:42.996442 39449 storage/replica_raftstorage.go:509  [s1,r1/1:/M{in-ax}] generated preemptive snapshot f012686b at index 17
I170801 23:59:43.002287 39449 storage/store.go:3486  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 36, log entries: 7, rate-limit: 2.0 MiB/sec, 3ms
I170801 23:59:43.003202 39311 storage/replica_raftstorage.go:718  [s3,r1/?:{-}] applying preemptive snapshot at index 17 (id=f012686b, encoded size=7179, 1 rocksdb batches, 7 log entries)
I170801 23:59:43.005117 39311 storage/replica_raftstorage.go:726  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I170801 23:59:43.008215 39449 storage/replica_command.go:3600  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3]
I170801 23:59:43.014334 39465 storage/raft_transport.go:451  raft transport stream to node 1 established
I170801 23:59:43.026589 39454 storage/replica.go:2937  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I170802 00:00:05.747612 39112 gossip/gossip.go:1218  [n1] node has connected to cluster via gossip
I170802 00:00:14.799291 39347 gossip/gossip.go:1218  [n3] node has connected to cluster via gossip
I170802 00:00:15.045404 39207 gossip/gossip.go:1218  [n2] node has connected to cluster via gossip
I170802 00:00:28.425412 39594 util/stop/stopper.go:471  quiescing; tasks left:
3      [async] storage.multiTestContextKVTransport: calling next replica
W170802 00:00:28.427654 39493 storage/replica.go:2593  [hb,s1,r1/1:/M{in-ax}] shutdown cancellation after 44.3s of attempting command [txn: 48a3c06e], BeginTransaction [/System/NodeLiveness/1,/Min), ConditionalPut [/System/NodeLiveness/1,/Min), EndTransaction [/System/NodeLiveness/1,/Min)
W170802 00:00:28.429994 39218 storage/node_liveness.go:379  [hb] slow heartbeat took 44.3s
I170802 00:00:28.430080 39218 storage/node_liveness.go:449  [hb] heartbeat result is ambiguous (server shutdown); retrying
W170802 00:00:28.430553 39218 storage/node_liveness.go:334  [hb] failed node liveness heartbeat: node unavailable; try another peer
I170802 00:00:28.430878 39594 util/stop/stopper.go:471  quiescing; tasks left:
2      [async] storage.multiTestContextKVTransport: calling next replica
W170802 00:00:28.434031 39494 storage/replica.go:2593  [hb,s1,r1/1:/M{in-ax}] shutdown cancellation after 44.2s of attempting command [txn: 1ab9b294], BeginTransaction [/System/NodeLiveness/2,/Min), ConditionalPut [/System/NodeLiveness/2,/Min), EndTransaction [/System/NodeLiveness/2,/Min)
W170802 00:00:28.434452 39308 storage/node_liveness.go:379  [hb] slow heartbeat took 44.2s
I170802 00:00:28.434525 39308 storage/node_liveness.go:449  [hb] heartbeat result is ambiguous (server shutdown); retrying
W170802 00:00:28.435039 39308 storage/node_liveness.go:334  [hb] failed node liveness heartbeat: node unavailable; try another peer
I170802 00:00:28.435550 39594 util/stop/stopper.go:471  quiescing; tasks left:
1      [async] storage.multiTestContextKVTransport: calling next replica
W170802 00:00:28.442184 39229 storage/replica.go:2593  [hb,s1,r1/1:/M{in-ax}] shutdown cancellation after 44.1s of attempting command [txn: dc9de9e4], BeginTransaction [/System/NodeLiveness/3,/Min), ConditionalPut [/System/NodeLiveness/3,/Min), EndTransaction [/System/NodeLiveness/3,/Min)
W170802 00:00:28.443234 39438 storage/node_liveness.go:379  [hb] slow heartbeat took 44.1s
I170802 00:00:28.443312 39438 storage/node_liveness.go:449  [hb] heartbeat result is ambiguous (server shutdown); retrying
W170802 00:00:28.443811 39438 storage/node_liveness.go:334  [hb] failed node liveness heartbeat: node unavailable; try another peer
W170802 00:00:28.446509 39477 storage/raft_transport.go:278  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W170802 00:00:28.449437 39477 storage/raft_transport.go:278  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W170802 00:00:28.455318 39467 storage/store.go:3239  [s3] raft error: node 1 claims to not contain store 1 for replica (n1,s1):1: store 1 was not found
W170802 00:00:28.456379 39465 storage/raft_transport.go:457  raft transport stream to node 1 failed: store 1 was not found
W170802 00:00:28.457121 39477 storage/raft_transport.go:278  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
I170802 00:00:28.460814 39603 storage/raft_transport.go:451  raft transport stream to node 1 established
W170802 00:00:28.465198 39598 storage/raft_transport.go:278  unable to accept Raft message from (n2,s2):?: no handler registered for (n1,s1):?
W170802 00:00:28.465913 39598 storage/raft_transport.go:278  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W170802 00:00:28.466125 39598 storage/raft_transport.go:278  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W170802 00:00:28.466929 39605 storage/store.go:3239  [s2] raft error: node 1 claims to not contain store 1 for replica (n1,s1):?: store 1 was not found
W170802 00:00:28.467014 39598 storage/raft_transport.go:278  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W170802 00:00:28.467287 39598 storage/raft_transport.go:278  unable to accept Raft message from (n2,s2):2: no handler registered for (n1,s1):1
W170802 00:00:28.501906 39469 storage/raft_transport.go:457  raft transport stream to node 3 failed: EOF
I170802 00:00:28.502229 39173 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:33153->127.0.0.1:46263: use of closed network connection
I170802 00:00:28.502492 39057 vendor/google.golang.org/grpc/grpclog/grpclog.go:60  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I170802 00:00:28.503654 39243 vendor/google.golang.org/grpc/grpclog/grpclog.go:60  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I170802 00:00:28.503913 39092 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:49093->127.0.0.1:34597: use of closed network connection
I170802 00:00:28.504367 39246 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:33153: operation was canceled"; Reconnecting to {127.0.0.1:33153 <nil>}
I170802 00:00:28.504478 39246 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  grpc: addrConn.transportMonitor exits due to: grpc: the connection is closing

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_log_queue_test.go
+++ b/pkg/storage/client_raft_log_queue_test.go
@@ -48,7 +48,7 @@ func TestRaftLogQueue(t *testing.T) {
 	// us in this test.
 	sc := storage.TestStoreConfig(nil)
 	sc.RaftTickInterval = math.MaxInt32
-	sc.RaftElectionTimeoutTicks = 1000000
+	setRaftElectionTimeoutTicks(&sc, 1000000)
 	mtc.storeConfig = &sc
 
 	defer mtc.Stop()

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -45,6 +45,14 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 )
 
+func setRaftElectionTimeoutTicks(sc *storage.StoreConfig, newVal int) {
+	sc.RaftElectionTimeoutTicks = newVal
+	// Recompute settings that are derived from RaftElectionTimeoutTicks.
+	// Not doing this has caused flakiness in the past (#17376).
+	sc.RangeLeaseActiveDuration, sc.RangeLeaseRenewalDuration = storage.RangeLeaseDurations(
+		storage.RaftElectionTimeout(sc.RaftTickInterval, sc.RaftElectionTimeoutTicks))
+}
+
 // mustGetInt decodes an int64 value from the bytes field of the receiver
 // and panics if the bytes field is not 0 or 8 bytes in length.
 func mustGetInt(v *roachpb.Value) int64 {
@@ -1169,7 +1177,7 @@ func TestStoreRangeCorruptionChangeReplicas(t *testing.T) {
 	}
 
 	// Don't timeout raft leader.
-	sc.RaftElectionTimeoutTicks = 1000000
+	setRaftElectionTimeoutTicks(&sc, 1000000)
 	mtc := &multiTestContext{
 		storeConfig: &sc,
 	}
@@ -1473,7 +1481,7 @@ func runReplicateRestartAfterTruncation(t *testing.T, removeBeforeTruncateAndReA
 	// Don't timeout raft leaders or range leases (see the relation between
 	// RaftElectionTimeoutTicks and rangeLeaseActiveDuration). This test expects
 	// mtc.stores[0] to hold the range lease for range 1.
-	sc.RaftElectionTimeoutTicks = 1000000
+	setRaftElectionTimeoutTicks(&sc, 1000000)
 	mtc := &multiTestContext{storeConfig: &sc}
 	defer mtc.Stop()
 	mtc.Start(t, 3)
@@ -1700,7 +1708,7 @@ func TestQuotaPool(t *testing.T) {
 	sc := storage.TestStoreConfig(nil)
 	// Suppress timeout-based elections to avoid leadership changes in ways
 	// this test doesn't expect.
-	sc.RaftElectionTimeoutTicks = 100000
+	setRaftElectionTimeoutTicks(&sc, 100000)
 	mtc := &multiTestContext{storeConfig: &sc}
 	mtc.Start(t, numReplicas)
 	defer mtc.Stop()
@@ -3269,7 +3277,7 @@ func TestTransferRaftLeadership(t *testing.T) {
 	// heavily loaded CPU is enough to reach the raft election timeout
 	// and cause leadership to change hands in ways this test doesn't
 	// expect.
-	sc.RaftElectionTimeoutTicks = 100000
+	setRaftElectionTimeoutTicks(&sc, 100000)
 	// This test can rapidly advance the clock via mtc.advanceClock(),
 	// which could lead the replication queue to consider a store dead
 	// and remove a replica in the middle of the test. Disable the

--- a/pkg/storage/client_split_test.go
+++ b/pkg/storage/client_split_test.go
@@ -1567,7 +1567,7 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 func TestLeaderAfterSplit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	storeConfig := storage.TestStoreConfig(nil)
-	storeConfig.RaftElectionTimeoutTicks = 1000000
+	setRaftElectionTimeoutTicks(&storeConfig, 1000000)
 	mtc := &multiTestContext{
 		storeConfig: &storeConfig,
 	}

